specs/agents/mobile/events.md

```diff
@@ -0,0 +1,53 @@
+## Mobile Events
+
+This document describes event used by the Mobile SDKs using the [OpenTelementry Event Api](https://github.com/open-telemetry/opentelemetry-specification/blob/0a4c6656d1ac1261cfe426b964fd63b1c302877d/specification/logs/event-api.md).
+Event names will be recording using the `event.name`
+
+
+### `ApplicationLaunch` 
+This event represents the occurrence of app entering the foreground or background or other application lifecycle states.
+
+#### Attributes
+| Name         | Type   | Values                 | Description                                        |
+|--------------|--------|------------------------|----------------------------------------------------|
+| `event.name` | String | `ApplicationLifecycle` | The name of the event                              |
```

Comment:
I agree with the naming convention from OTel.

The concepts of `foreground` and `background` also apply to Android, however, I like it better the idea of having separate event names. But to be honest, I'm not sure if we need these events? This seems to have some overlapping with the last one (`application_opened`) which I remember was what WF was mentioning, I'm not sure if they also want to tell when their app is closed and/or backgrounded 🤔

Reply:
I'm a bit confused tbh, I'm not sure if it's that we're trying to address multiple use cases and probably I'm not aware of them all. So I remember that, one of the events that WF needs is to tell when the app has been opened (either from scratch or brought back to the foreground) and that's what I thought `application_opened` was for, and then maybe these other events were useful to tell a bit more of the story, as a complimentary feature for more in-depth use cases (and having a bit of natural overlapping with application_opened). But now that you mention that `for iOS application_foreground would be more appropriate than application_opened`, that's where I got confused because I thought we were aiming for having the same event names for both, but instead it seems like you've added `application_opened` thinking of android's use cases?

The `foreground` and `background` concepts are familiar in Android too, so I think in that sense we could find some common namings. I also think that in some cases we might need to provide platform-specific names when there's no common ground, such as what we did with the [app's metrics](https://github.com/elastic/apm/blob/main/specs/agents/mobile/metrics.md#application-metrics), but for common stuff, such as simply telling when the app has been opened, I believe it might be better if we use the same namings because I think it might make things more intuitive for our customers to query these scenarios for both platforms. At least with WF's current solution for example, it seems like they treat "app opens" as something universal across mobile platforms which seems handy for them.

Reply:
> I'm a bit confused tbh, I'm not sure if it's that we're trying to address multiple use cases and probably I'm not aware of them all
I'm trying to suss that out as well. I think depending on the use case, we might want a single event with multiple states (as I originally proposed) or multiple events.
What sort of value do tracking lifecycle events have:

-  context for a session view: I see the user transitioned from the background to the foreground several times while using the app .
- a count of application usage: how many app launches did we experience in a given period
- debugging : Why did my app crash calling this method while the app was background, and not while my app was in the foreground

app launches are not a good metric for tracking application usage. Unique sessions, and unique users are much better usage metrics for overall app uses, each for different purposes : 
- You can compare the number of unique users to the number of unique crashes, and show the percentage of users that are experiencing crashes.
- Unique sessions is a more refined usage metric over app launches: a user can open and close the app repeatedly to drive an app launch count up giving an inflated sense of usage: this sort of usage inflation doesn't occur using session counts.

Reply:
> The foreground and background concepts are familiar in Android too, so I think in that sense we could find some common namings. I also think that in some cases we might need to provide platform-specific names when there's no common ground, such as what we did with the [app's metrics](https://github.com/elastic/apm/blob/main/specs/agents/mobile/metrics.md#application-metrics), but for common stuff, such as simply telling when the app has been opened, I believe it might be better if we use the same namings because I think it might make things more intuitive for our customers to query these scenarios for both platforms.

This is a fair point, but I'm not sure there's as much common ground as it first appears. Android has several different processes associated with [application lifecycle](https://developer.android.com/guide/components/activities/process-lifecycle) that might not fit neatly into the `open-closed`/`launched-terminated` events proposed. I'm not as familiar with the android lifecycles, but I've run into issues with `server` processes fitting into similarly defined events.